llama-index-integrations/vector_stores/llama-index-vector-stores-databricks/llama_index/vector_stores/databricks/base.py

```diff
@@ -216,8 +219,19 @@ def add(
         ids = []
         for node in nodes:
             node_id = node.node_id
-            metadata = node_to_metadata_dict(node, remove_text=True, flat_metadata=True)
 
+            # unroll metadata in node_content
+            node_metadata = node_to_metadata_dict(
+                node, remove_text=True, flat_metadata=True
+            )
+            node_content = json.loads(node_metadata.get("_node_content", {}))
+            metadata = node_content.get("metadata", {})
```

Comment:
I'm a bit confused here. What the purpose of these conversions? We don't we access the attribute of `node` directly?

```
metadata = node.metadata or {}
node_info = {key: getattr(key) for key in self._node_info_columns}
```
It look like the json serde here is unnecessary overhead.